iocore/net/SSLUtils.cc

```diff
@@ -1316,6 +1316,13 @@ SSLMultiCertConfigLoader::init_server_ssl_ctx(std::vector<X509 *> &cert_list, co
     // Session tickets are enabled by default. Disable if explicitly requested.
     if (sslMultCertSettings->session_ticket_enabled == 0) {
       SSL_CTX_set_options(ctx, SSL_OP_NO_TICKET);
+#if OPENSSL_VERSION_NUMBER >= 0x10101000L && !defined(LIBRESSL_VERSION_NUMBER)
+      // If TLSv1.3 is being used we need to set the number of tickets to 0 to
+      // properly disable TLS tickets for TLSv1.3 connections
+      if (!(params->ssl_ctx_options & SSL_OP_NO_TLSv1_3)) {
```

Comment:
I don't think this is the best way to do the check. From my testing, if we use OpenSSL 1.1.1, then because of ATS does a non-blocking handshake, the first pass will always be TLS v1.3, and then the correct TLS version will then be used on the future communications. Maybe `SSL_set_num_tickets` is a better choice?

Reply:
Also, it seems like it breaks session resumption on TLS v1.3 with this PR.
`openssl s_client -connect 127.0.0.1:4443 -sess_out ./sess.dat -sess_in ./sess.dat -tls1_3`
returns
`New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384`
if I set the `proxy.config.ssl.server.session_ticket.enable` to `0`. Resumption still works if TLS v1.2 is used with that option set to `0`.

Reply:
for me it's already broken before this PR sadly